third_party/terraform/resources/resource_dataflow_job.go

```diff
@@ -139,15 +145,17 @@ func resourceDataflowJobCreate(d *schema.ResourceData, meta interface{}) error {
 	}
 
 	params := expandStringMap(d, "parameters")
+	labels := expandStringMap(d, "labels")
 
 	env := dataflow.RuntimeEnvironment{
-		MaxWorkers:          int64(d.Get("max_workers").(int)),
-		Network:             d.Get("network").(string),
-		ServiceAccountEmail: d.Get("service_account_email").(string),
-		Subnetwork:          d.Get("subnetwork").(string),
-		TempLocation:        d.Get("temp_gcs_location").(string),
-		MachineType:         d.Get("machine_type").(string),
-		Zone:                zone,
+		MaxWorkers:           int64(d.Get("max_workers").(int)),
+		Network:              d.Get("network").(string),
+		ServiceAccountEmail:  d.Get("service_account_email").(string),
+		Subnetwork:           d.Get("subnetwork").(string),
+		TempLocation:         d.Get("temp_gcs_location").(string),
+		MachineType:          d.Get("machine_type").(string),
+		AdditionalUserLabels: labels,
```

Comment:
You should also be able to d.Set labels on read: https://cloud.google.com/dataflow/docs/reference/rest/v1b3/projects.locations.jobs